docs/languages/csharp/expression-trees.md

```diff
@@ -0,0 +1,132 @@
+`MoreSecretSauce()`, that binary addition operator may be a method call to an
+overridden addition operator.
+
+Despite this perceived complexity, the expression above creates a tree strcuture
+that can be navigated as easily as the first sample. You can keep traversing
+child nodes to find leaf nodes in the expression. Parent nodes will have
+references to their children, and each node has a property that describes
+what kind of node it is.
+
+The structure of an expression tree is very consistent. Once you've learned
+the basics, you can understand even the most complex code. The elegance in
+the data structure explains how the C# compiler can analyze the most complex
+C# programs and create proper output from that complicated source code.
+
+Once you become familiar with the structure of expression trees, you will
+find that knowledge you've gained quickly enables you to work with many
+more and more advanced scenarios. There is incredible power to expression
+trees.
+## Uses for Expression Trees
+You've likely already worked with Expression Trees in libraries that
+you use. One of the most common is Entity Framework (link?). When you
+write queries against Entity Framework sets, you define expression trees.
+The code inside Entity Framework traverses those expression trees, and
+translates the C# code that you wrote into parameterized SQL queries.
+Regardless of the complexity of the query expressions you create, whether
+you use the query syntax, or the fluent method call syntax, Entity Framework
+translates the expression tree into SQL statements.
+
+There are some limitations to what Entity Framework can translate. In general,
+Entity Framework doesn't translate arbitrary method calls in query expressions.
+But, most other expressions are correctly converted. (Many method calls where
+the receiver is a well known .NET Framework type are converted. Consult the
+Entity Framework documentation for more details.)
+
+That's one example of the power of expression trees. The code you write isn't
+compiled into IL, rather, it is stored in an expression tree, which is a 
+data structure you can examine. That data structure
+can be understood by a library. That library can then translate your algorithms
+to another language, or another environment.
+
+In addition to translating algorithms to execute in other environments,
+expression trees can be used to make it easier to write algorithms that inspect
+code before executing it. You can write a method whose arguments are expressions
+and then examine those expressions before executing the code. The Expression Tree
+is a full representation of the code: you can see values of any sub-expression.
+You can see method and property names. You can see the value of any constant expressions.
+You can also convert an expression tree into an executable delegate, and execute the
+code.
+
+The one thing you can't do is modify an expression tree.  Expression Trees are immutable
```

Comment:
This paragraph may go better in another section which describes the data structure a bit more.  When I'm reading this section, I'm looking for examples of Expression Tree use so that I can be convinced that they're a good tool to use.